src/main/java/uk/gov/hmcts/reform/em/hrs/config/security/EmServiceAuthFilter.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.em.hrs.config.security;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EmServiceAuthFilter extends AbstractPreAuthenticatedProcessingFilter {
+
+    public static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    public static final String NOT_APPLICABLE = "N/A";
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmServiceAuthFilter.class);
+
+    private final List<String> authorisedServices;
+
+    private final AuthTokenValidator authTokenValidator;
+
+    public EmServiceAuthFilter(AuthTokenValidator authTokenValidator, List<String> authorisedServices) {
+
+        this.authTokenValidator = authTokenValidator;
+        if (authorisedServices == null || authorisedServices.isEmpty()) {
```

Comment:
You could use CollectionUtils.isEmpty(). Just one check as CollectionUtils.isEmpty() method is null safe.